City-of-Helsinki/drupal-helfi-platform-config
# UHF-7029: Created helfi calculator module and added paragraph to stan…

# TASKNAME [UHF-7029](https://helsinkisolutionoffice.atlassian.net/browse/UHF-7029)

POC for calculator module

## What was done
* Created a helfi_calculator module which adds calculator paragraph and a settings page for it

## How to install
* Make sure your instance is up and running on latest dev branch.
    * `git pull origin dev`
    * `make fresh`
* Update the Helfi Platform config
    * `composer require drupal/helfi_platform_config:dev-UHF-7029-helfi-calculator`
    * `drush en -y helfi_calculator`
* Run `make drush-updb drush-cr`

## How to test
* After installing the module, go check that the paragraph was created `/fi/admin/structure/paragraphs_type/calculator`
    * The paragraph should have only a "Disabled" value here `/fi/admin/structure/paragraphs_type/calculator/fields/paragraph.calculator.field_calculator/storage`
* Go to the calculators settings page `tools > calculator settings`
    * All of the calculators should be off and they should have as value {}
* Try enabling some of the calculators (depending on your local environment cache settings you might have to flush all caches form the drop in admin tool bar after saving the configs in order for the changes to appear)
* After enabling some, go check out the paragraph field settings, it should now have the original "disabled" and the ones that were just checked on
    * If you disable one that was enabled, it should disappear from the paragraph options and still leave the disabled one as the first one 

Go through the code and check if it makes sense. Is everything logical and is there something that should be done in a different way?
